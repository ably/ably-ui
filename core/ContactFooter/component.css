@layer components {
  .ui-contact-footer {
    background-size: 100% 100%;
    background-position: right center;
    @apply w-full bg-gradient-active-orange;
  }

  .ui-contact-footer-box {
    @apply p-24 sm:p-32 xl:p-40 bg-white flex flex-col justify-between rounded-sm;
  }
}


/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIndlYnBhY2s6Ly9BYmx5VWkuQ29yZS5bbmFtZV0vLi9zcmMvY29yZS9Db250YWN0Rm9vdGVyL2NvbXBvbmVudC5jc3MiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBQUE7RUFDRTtJQUNFLDBCQUEwQjtJQUMxQixpQ0FBaUM7SUFDakMsdUNBQXVDO0VBQ3pDOztFQUVBO0lBQ0UsNkVBQTZFO0VBQy9FO0FBQ0YiLCJmaWxlIjoiY29yZS9Db250YWN0Rm9vdGVyL2NvbXBvbmVudC5jc3MiLCJzb3VyY2VzQ29udGVudCI6WyJAbGF5ZXIgY29tcG9uZW50cyB7XG4gIC51aS1jb250YWN0LWZvb3RlciB7XG4gICAgYmFja2dyb3VuZC1zaXplOiAxMDAlIDEwMCU7XG4gICAgYmFja2dyb3VuZC1wb3NpdGlvbjogcmlnaHQgY2VudGVyO1xuICAgIEBhcHBseSB3LWZ1bGwgYmctZ3JhZGllbnQtYWN0aXZlLW9yYW5nZTtcbiAgfVxuXG4gIC51aS1jb250YWN0LWZvb3Rlci1ib3gge1xuICAgIEBhcHBseSBwLTI0IHNtOnAtMzIgeGw6cC00MCBiZy13aGl0ZSBmbGV4IGZsZXgtY29sIGp1c3RpZnktYmV0d2VlbiByb3VuZGVkLXNtO1xuICB9XG59XG4iXSwic291cmNlUm9vdCI6IiJ9*/