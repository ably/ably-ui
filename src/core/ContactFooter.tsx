import React, { useEffect } from "react";

import Icon from "./Icon";
import _absUrl from "./url-base.js";
import toggleChatWidget from "./hubspot-chat-toggle";

type ContactFooterProps = {
  urlBase: string;
};

const ContactFooter = ({ urlBase }: ContactFooterProps) => {
  useEffect(() => toggleChatWidget({ dataId: "contact-footer" }), []);
  const absUrl = (path: string) => _absUrl(path, urlBase);

  return (
    <div
      className="ui-contact-footer font-sans antialiased"
      data-id="contact-footer"
    >
      <div className="w-full bp-lg max-w-screen-xl mx-auto py-64 grid grid-cols-1 md:grid-cols-3 ui-grid-gap ui-grid-px">
        <div className="ui-contact-footer-box">
          <Icon
            name="icon-display-live-chat"
            size="3rem"
            additionalCSS="block mb-16"
          />
          <div>
            <div className="ui-text-h3 mb-24">Live Chat</div>
            <p className="ui-text-p1">
              Reach out team of experts over chat powered by Ably.
            </p>
          </div>
          <button
            type="button"
            className="ui-btn-secondary self-start mt-16"
            disabled
            data-id="open-chat-widget"
            data-enabled-label="Start a live chat"
            data-disabled-label="Live chat unavailable"
          >
            Live chat unavailable
          </button>
        </div>

        <div className="ui-contact-footer-box">
          <Icon
            name="icon-display-call-mobile"
            size="3rem"
            additionalCSS="block mb-16"
          />
          <div className="flex-grow">
            <div className="ui-text-h3 mb-24">Call us</div>
            <p className="ui-text-p1">
              <span className="block">
                <strong className="font-bold">+1 877 434 5287</strong> (USA,
                toll free)
              </span>
              <span className="block">
                <strong className="font-bold">+44 20 3318 4689</strong> (UK)
              </span>
            </p>
          </div>
        </div>

        <div className="ui-contact-footer-box">
          <Icon
            name="icon-display-tech-account-comms"
            size="3rem"
            additionalCSS="block mb-16"
          />
          <div>
            <div className="ui-text-h3 mb-24">
              Technical and account support
            </div>
            <p className="ui-text-p1">
              We&apos;re standing by to help with any questions or code.
            </p>
          </div>
          <a
            className="ui-btn-secondary self-start p-btn mt-16"
            href={absUrl("/support")}
          >
            Get support now
          </a>
        </div>
      </div>
    </div>
  );
};

export default ContactFooter;
