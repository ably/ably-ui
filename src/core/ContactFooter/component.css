@layer components {
  .ui-contact-footer {
    background: var(--gradient-active-orange);
    background-size: 100% 100%;
    background-position: right center;
    @apply w-full;
  }
}
