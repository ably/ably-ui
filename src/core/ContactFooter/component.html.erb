<div class="ui-contact-footer font-sans antialiased" data-id="contact-footer">
  <div class="w-full bp-lg max-w-screen-xl mx-auto py-64 grid grid-cols-1 md:grid-cols-3 ui-grid-gap ui-grid-px">
    <div class="ui-contact-footer-box">
      <%= render(AblyUi::Core::Icon.new(name: "icon-display-live-chat", size: "3rem", additional_css: "block mb-16")) %>
      <div>
        <div class="text-h3 mb-24">Live Chat</div>
        <p class="text-p1 font-light">Reach out team of experts over chat powered by Ably.</p>
      </div>
      <button type="button" class="ui-btn-secondary self-start mt-16" data-id="open-chat-widget" disabled>Live chat unavailable</button>
    </div>

    <div class="ui-contact-footer-box">
      <%= render(AblyUi::Core::Icon.new(name: "icon-display-call-mobile", size: "3rem", additional_css: "block mb-16")) %>
      <div class="flex-grow">
        <div class="text-h3 mb-24">Call us</div>
        <p class="text-p1 font-light">
          <span class="block"><strong class="text-p1 font-medium">+1 877 434 5287</strong> (USA, toll free) </span>
          <span class="block"><strong class="text-p1 font-medium">+44 20 3318 4689</strong> (UK) </span>
        </p>
      </div>
    </div>

    <div class="ui-contact-footer-box">
      <%= render(AblyUi::Core::Icon.new(name: "icon-display-tech-account-comms", size: "3rem", additional_css: "block mb-16")) %>
      <div>
        <div class="text-h3 mb-24">Technical and account support</div>
        <p class="text-p1 font-light">We&apos;re standing by to help with any questions or code.</p>
      </div>
      <%= link_to 'Get support now', abs_url("/support"), class: "ui-btn-secondary self-start p-btn mt-16" %>
    </div>
  </div>
</div>
