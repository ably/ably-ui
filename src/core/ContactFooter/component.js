import toggleChatWidget from "../hubspot-chat-toggle";
export default () => toggleChatWidget({ dataId: "contact-footer" });
