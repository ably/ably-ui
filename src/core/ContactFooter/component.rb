module AblyUi
  module Core
    class ContactFooter < ViewComponent::Base
    end
  end
end
