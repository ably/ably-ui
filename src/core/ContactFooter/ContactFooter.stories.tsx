import ContactFooter from "./component.jsx";

export default {
  title: "Components/Contact Footer",
  component: ContactFooter,
  parameters: {
    layout: "fullscreen",
  },
};

export const Default = {};
