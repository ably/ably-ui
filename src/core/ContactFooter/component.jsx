import React, { useEffect } from "react";

import "./component.css";
import toggleChatWidget from "./component.js";

export default function ContactFooter() {
  const listItemClasses = "p-24 bg-white flex flex-col justify-between";
  const svgicon = (id) => (
    <svg className="w-48 h-48 block mb-16">
      <use xlinkHref={"#" + id}></use>
    </svg>
  );

  useEffect(() => toggleChatWidget(), []);

  return (
    <div className="ui-contact-footer font-sans" data-id="contact-footer">
      <div className="w-full bp-lg max-w-screen-xl mx-auto py-64 grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 ui-grid-gap ui-grid-px">
        <div className={listItemClasses}>
          {svgicon("sprite-live-chat")}
          <div>
            <div className="text-h3 mb-24">Live Chat</div>
            <p className="text-p1 font-light">Reach out team of experts over chat powered by Ably.</p>
          </div>
          <button type="button" className="ui-btn-secondary self-start p-btn-small mt-16" data-id="open-chat-widget">
            Start a live chat
          </button>
        </div>

        <div className={listItemClasses}>
          {svgicon("sprite-call-mobile")}
          <div className="flex-grow">
            <div className="text-h3 mb-24">Call us</div>
            <p className="text-p1 font-light">
              <span className="block">
                <strong className="text-p1 font-medium">+1 877 434 5287</strong> (USA, toll free)
              </span>
              <span className="block">
                <strong className="text-p1 font-medium">+44 20 3318 4689</strong> (UK)
              </span>
            </p>
          </div>
        </div>

        <div className={listItemClasses}>
          {svgicon("sprite-tech-account-comms")}
          <div>
            <div className="text-h3 mb-24">Technical and account support</div>
            <p className="text-p1 font-light">We're standing by to help with any questions or code.</p>
          </div>
          <a className="ui-btn-secondary self-start p-btn-small mt-16" href="https://google.com">
            Get support now
          </a>
        </div>
      </div>
    </div>
  );
}
