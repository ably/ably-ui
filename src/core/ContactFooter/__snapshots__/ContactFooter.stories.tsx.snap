// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Contact Footer Default smoke-test 1`] = `
<div class="ui-contact-footer font-sans antialiased"
     data-id="contact-footer"
>
  <div class="w-full bp-lg max-w-screen-xl mx-auto py-64 grid grid-cols-1 md:grid-cols-3 ui-grid-gap ui-grid-px">
    <div class="ui-contact-footer-box">
      <svg class="block mb-16"
           style="width: 3rem; height: 3rem;"
      >
        <use xlink:href="#sprite-icon-display-live-chat">
        </use>
      </svg>
      <div>
        <div class="ui-text-h3 mb-24">
          Live Chat
        </div>
        <p class="ui-text-p1">
          Reach out team of experts over chat powered by Ably.
        </p>
      </div>
      <button type="button"
              class="ui-btn-secondary self-start mt-16"
              disabled
              data-id="open-chat-widget"
              data-enabled-label="Start a live chat"
              data-disabled-label="Live chat unavailable"
      >
        Live chat unavailable
      </button>
    </div>
    <div class="ui-contact-footer-box">
      <svg class="block mb-16"
           style="width: 3rem; height: 3rem;"
      >
        <use xlink:href="#sprite-icon-display-call-mobile">
        </use>
      </svg>
      <div class="flex-grow">
        <div class="ui-text-h3 mb-24">
          Call us
        </div>
        <p class="ui-text-p1">
          <span class="block">
            <strong class="font-bold">
              +1 877 434 5287
            </strong>
            (USA, toll free)
          </span>
          <span class="block">
            <strong class="font-bold">
              +44 20 3318 4689
            </strong>
            (UK)
          </span>
        </p>
      </div>
    </div>
    <div class="ui-contact-footer-box">
      <svg class="block mb-16"
           style="width: 3rem; height: 3rem;"
      >
        <use xlink:href="#sprite-icon-display-tech-account-comms">
        </use>
      </svg>
      <div>
        <div class="ui-text-h3 mb-24">
          Technical and account support
        </div>
        <p class="ui-text-p1">
          We're standing by to help with any questions or code.
        </p>
      </div>
      <a class="ui-btn-secondary self-start p-btn mt-16"
         href="/support"
      >
        Get support now
      </a>
    </div>
  </div>
</div>
`;
